pkg/kubelet/token/token_manager.go

```diff
@@ -187,7 +189,8 @@ func (m *Manager) requiresRefresh(tr *authenticationv1.TokenRequest) bool {
 	if now.After(exp.Add(-1*time.Duration((*tr.Spec.ExpirationSeconds*20)/100)*time.Second - jitter)) {
 		return true
 	}
-	return false
+	// Require a refresh if the issued time is later than the system time.
+	return now.Before(iat)
```

Comment:
/hold

if we encounter this, this means we have clock skew issues (either the kubelet clock is wrong, or the kube-apiserver clock was wrong when it issued the token)

it is not clear that requesting a new token will make the situation better. In the worst case this will hotloop token requests from the kubelet

Reply:
This change is not aimed at resolving the clock skew issues, but at ensuring that invalid tokens generated during the period after the clock skew issues (may later than the correct time) has been fixed can be effectively refreshed.

Otherwise, even after the time correction, all actions relying on tokens to access the API server will be rejected, and the related services will never be able to recover.

Reply:
1. liveness prober failure can make container recreate, but token will not re-generate.
2. removing pod and recreating a new one can fix this promblem, but it's can be very heavy for stateful pod. Because just one API in time skew may cause this problem to occur in most Pods in the entire cluster.

In my humble opinion, even the worst case (hotloop and overload API) can happen,  but a dead APIserver can be better than a bad APIServer. Because if this promblem occured, pod can not access apiserver because of invalid token too.  If we check-in this fix, the cluster admin just fix the time issues and wait APIServer come back online. But if we not check-in this fix, the cluster admin additionally need to recreate the most of pods in the cluster. 

furthermore, A broken control-plane usually will not break applications running on the platform, but restart all of pods are much more dangerous.

Reply:
> if we encounter this, this means we have clock skew issues (either the kubelet clock is currently wrong, or the kube-apiserver clock was wrong when it issued the token, or both)
> 
> it is not clear that requesting a new token will make the situation better. In the worst case this will hotloop token requests from the kubelet

This is still my assessment of the situation ... if we encounter this, it is clear a misconfiguration is or was present, it is not clear the misconfiguration is resolved, and taking action on the node side to try to renew the token can actually hurt the system more than it helps.